File: D12_121.Best Time to Buy and Sell Stock/CaiChen/121.py
class Solution(object):
    def maxProfit(self, prices):
        """
        :type prices: List[int]
        :rtype: int
        """
        if len(prices) < 2:
            return 0
        currMax = 0
        res = 0
        for i in xrange(1, len(prices)):
            currMax = max((prices[i] - prices[i - 1]), (prices[i] - prices[i - 1] + currMax))
            res = max(currMax, res)
        return res

File: D12_121.Best Time to Buy and Sell Stock/CaiChen/121.java
public class Solution 
{
    public int maxProfit(int[] prices) 
    {
        if(prices == null || prices.length < 2)
            return 0;
        int max = 0, currMax = 0;
        for(int i = 1; i < prices.length; i++)
        {
            currMax = Math.max((prices[i] - prices[i - 1]), (prices[i] - prices[i - 1] + currMax));
            max = Math.max(currMax, max);
        }
        return max;
    }
}